feat(proxy): support absolute RPM/TPM in priority_reservation

AlexsanderHamir opened the pull request:
## Title

feat(proxy): support absolute RPM/TPM in priority_reservation

## Relevant issues

<!-- e.g. "Fixes #000" -->

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [ ] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

<!-- Select the type of Pull Request -->
<!-- Keep only the necessary ones -->

🆕 New Feature

## Changes

• Extended `priority_reservation` to accept both float and dict values (absolute RPM/TPM support)
• Added `convert_priority_to_percent` helper in `rate_limiter_utils.py`
• Updated `dynamic_rate_limiter` and `dynamic_rate_limiter_v3` to use the new conversion logic

## Local Results

<img width="1155" height="270" alt="Screenshot 2025-10-22 at 2 38 34 PM" src="https://github.com/user-attachments/assets/b30a89a9-5787-406b-81f3-a6c69092270e" />

AlexsanderHamir committed:
feat(proxy): support absolute RPM/TPM in priority_reservation

Allow priority reservations as absolute values instead of percentages:
- Float: {'prod': 0.75} (75%, existing)
- RPM: {'prod': {'type': 'rpm', 'value': 750}}
- TPM: {'prod': {'type': 'tpm', 'value': 750000}}

Added _convert_to_percent() that converts absolute values to percentages
based on model capacity. Fully backward compatible.

AlexsanderHamir commented:
@ishaan-jaff, can this be any simpler ? do I need to do anything different ?

ishaan-jaff reviewed (changes requested):
- Can you add screenshots of this working when set on config.yaml and starting up 
- Can you update this doc https://docs.litellm.ai/docs/proxy/dynamic_rate_limit#beta-set-priority--reserve-quota

AlexsanderHamir committed:
feat(types): convert priority_reservation Dict to TypedDict

Add PriorityReservationDict TypedDict to replace generic Dict type in priority_reservation configuration.

Changes:
- Add PriorityReservationDict to litellm/types/utils.py
- Update convert_priority_to_percent() signature in rate_limiter_utils.py
- Update litellm.priority_reservation type annotation in __init__.py

Improves IDE autocomplete and type checking for priority reservation configs.

AlexsanderHamir committed:
docs: update dynamic rate limiter priority reservation docs
